controlplane/controllers/scale.go

```diff
@@ -159,6 +160,27 @@ func (r *CK8sControlPlaneReconciler) scaleDownControlPlane(
 	}
 	**/
 
+	clusterObjectKey := util.ObjectKey(cluster)
+	workloadCluster, err := r.managementCluster.GetWorkloadCluster(ctx, clusterObjectKey)
+	if err != nil {
+		logger.Error(err, "failed to create client to workload cluster")
+		return ctrl.Result{}, errors.Wrapf(err, "failed to create client to workload cluster")
+	}
+
+	authToken, err := token.Lookup(ctx, r.Client, clusterObjectKey)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to lookup auth token: %w", err)
+	}
+
+	if authToken == nil {
+		return ctrl.Result{}, fmt.Errorf("auth token not yet generated")
+	}
+
+	err = workloadCluster.RemoveMachineFromCluster(ctx, machineToDelete, *authToken)
+	if err != nil {
```

Comment:
otherwise err trickles down to the rest of the code, let's avoid this

```suggestion
	if err := workloadCluster.RemoveMachineFromCluster(ctx, machineToDelete, *authToken); err != nil {
```